dev/release/verify-release-candidate.sh

```diff
@@ -263,7 +263,6 @@ test_yum() {
                     "almalinux:8" \
                     "amazonlinux:2023" \
                     "quay.io/centos/centos:stream9" \
-                    "quay.io/centos/centos:stream8" \
                     "centos:7"; do
```

Comment:
Why do we keep `centos:7` here? It's EOL too, isn't it?

Reply:
https://github.com/apache/arrow/issues/40735#issuecomment-2787159983

Voltron Data wants to keep it.

Reply:
I see.

Reply:
If we are targeting v22 for this PR, we probably can remove `centos:7` too as the request was to keep it for the v21 release cycle. @ianmcook @assignUser fo you know if we can remove it for v22